compiler/src/dotty/tools/dotc/transform/IsInstanceOfEvaluator.scala

```diff
@@ -147,7 +185,7 @@ class IsInstanceOfEvaluator extends MiniPhaseTransform { thisTransformer =>
 
           val inMatch = s.qualifier.symbol is Case
```

Comment:
This is not a correct discrimination for isInstanceOf's used in pattern matching! It would also cover case objects and enum cases. We need a more robust check. Maybe have a special, second `isInstanceOf` symbol?